### PR TITLE
Fixing issues 215 / 244 due to xjc-ri changes from 2.3.4

### PR DESCRIPTION
**Pull request by laurentschoelens**
Mainly the goal here is to fix :

- https://github.com/highsource/jaxb-tools/issues/215 : reactivate the MAVEN_JAXB2_PLUGIN-77 with fix
- https://github.com/highsource/jaxb-tools/issues/244 : add a new test project (jt-244) that validate the fix

Also done some cosmetic changes to :
- maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ClasspathCatalogResolver.java
- maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/MavenCatalogResolver.java

### CODE REVIEW

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java [hunk 1]
```diff
@@ -12,33 +18,51 @@ public class ReResolvingEntityResolverWrapper implements EntityResolver {
...
+		boolean fileExistsResolvedSystemId = checkFileExists(resolvedSystemId);
+		return !StringUtils.isEmpty(systemId) && fileExistsSystemId ? systemId : fileExistsResolvedSystemId ? resolvedSystemId : systemId;
+	}
+
+	private static boolean checkFileExists(String sId) {
+		try {
+			URI uriSystemId = new URI(sId);
+			if ("file".equals(uriSystemId.getScheme())) {
+				if (!Files.exists(Paths.get(uriSystemId))) {
+					// resolved file does not exist, warn and let's continue with original systemId
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java:59]: Should we log a WARN here if user indicates a file:// but the file does not exist? Seems like a swallowed error condition

**Review by mattrpav — CHANGES_REQUESTED** [review]

**Comment by laurentschoelens** [comment]
- [x] Adding new Interface to allow passing `getLog()` to CatalogResolver instance
- [x] Changed System.out / System.err to appropriate `log.xxx` methods
- [x] Checked on test projects that only appropriate logs get logged by default and all logs with `-X -e` flags

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java [hunk 2]
```diff
@@ -12,33 +18,51 @@ public class ReResolvingEntityResolverWrapper implements EntityResolver {
...
 	}
 
 	@Override
 	public InputSource resolveEntity(String publicId, String systemId)
 			throws SAXException, IOException {
-		// System.out.println(MessageFormat.format("Resolving publicId [{0}], systemId [{1}].",
-		// publicId, systemId));
-		final InputSource resolvedInputSource = this.entityResolver
-				.resolveEntity(publicId, systemId);
+		// System.out.println(MessageFormat.format("Resolving publicId [{0}], systemId [{1}].", publicId, systemId));
```
[the remark this answers is not in the document]
Comment (laurentschoelens) [reply maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java]: Done

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java (hunk 1 above)
Comment (laurentschoelens) [reply maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingEntityResolverWrapper.java:64]: Done too

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingInputSourceWrapper.java [hunk 3]
```diff
@@ -27,22 +33,34 @@ public Reader getCharacterStream() {
...
+					getPublicId(), getSystemId());
+			if (resolvedEntity == null) {
+				resolvedEntity = this.entityResolver.resolveEntity(resolvedPublicId, resolvedSystemId);
+			}
+			if (resolvedEntity == null) {
+				return null;
+			} else {
+				return resolvedEntity.getCharacterStream();
+			}
+		} catch (IOException ioex) {
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingInputSourceWrapper.java:57]: Should we log warn here?

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingInputSourceWrapper.java [hunk 4]
```diff
@@ -27,22 +33,34 @@ public Reader getCharacterStream() {
...
+			return null;
+		} catch (SAXException saxex) {
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/resolver/tools/ReResolvingInputSourceWrapper.java:59]: Should we log warn here?

**Review by mattrpav — APPROVED** [review]

**Pull request merged**